solarforecastarbiter/metrics/tests/test_preprocessing.py

```diff
@@ -237,6 +237,50 @@ def test_resample_and_align_timezone(site_metadata, interval_label, tz,
                                   check_categorical=False)
 
 
+def test_resample_and_align_with_ref(
+        single_forecast_observation_reffx):
+    tz = 'UTC'
+    fx_obs = single_forecast_observation_reffx
+    fx_series = THREE_HOUR_SERIES
+    ref_series = THIRTEEN_10MIN_SERIES
```

Comment:
I intended to only support reference forecasts that had the same `freq` as the forecast, so I think this should raise an exception. But I don't think we'd hit this via the API/dashboard though because of the way intervals are enforced in the API upload.

Reply:
hmm, should maybe add that to the datamodel too

Reply:
add = add to documentation?

On further thought, it seems hard to test in the datamodel or here. We don't require observations to have a fixed frequency since they can have missing data.

The issue with the test is that I think you have an unnatural pairing of the observation data with the observation metadata.

Reply:
Ah, you actually do raise the error, this test series just isn't consistent with the ref Forecast datamodel

Reply:
I meant add an interval compatibility test to the datamodel for ForecastObservation with a reference forecast. We already check that the observation is compatible with the forecast, but don't check the reference forecast at all

Reply:
I'm ok leaving that for a follow up issue. I think `preprocessing` needs some refactoring to make it more flexible, so that might not actually happen.